stratum/hal/lib/barefoot/bf_pal_wrapper.cc

```diff
@@ -280,6 +280,47 @@ ::util::Status BFPalWrapper::PortLoopbackModeSet(int unit, uint32 port_id,
   return ::util::OkStatus();
 }
 
+::util::Status BFPalWrapper::PortIdFromPortKeyGet(int unit,
+                                                  const PortKey& port_key,
+                                                  uint32* sdk_port_id) {
+  const int port = port_key.port;
+  CHECK_RETURN_IF_FALSE(port >= 0)
+      << "Port ID must be non-negative. Attempted to get port " << port
+      << " on dev " << unit << ".";
+
+  // PortKey uses three possible values for channel:
+  //     > 0: port is channelized (first channel is 1)
+  //     0: port is not channelized
+  //     < 0: port channel is not important (e.g. for port groups)
+  // BF SDK expects the first channel to be 0
+  //     Convert base-1 channel to base-0 channel if port is channelized
+  //     Otherwise, port is already 0 in the non-channelized case
+  const int channel =
+      (port_key.channel > 0) ? port_key.channel - 1 : port_key.channel;
+  CHECK_RETURN_IF_FALSE(channel >= 0)
+      << "Channel must be set for port " << port << " on dev " << unit << ".";
+
+  char port_string[MAX_PORT_HDL_STRING_LEN];
+  int r =
+      snprintf(port_string, MAX_PORT_HDL_STRING_LEN, "%d/%d", port, channel);
```

Comment:
```suggestion
      snprintf(port_string, sizeof(port_string), "%d/%d", port, channel);
```

See: any other usage of `snprintf`